src/execution_plan/optimizations/reduce_traversal.c

```diff
@@ -0,0 +1,109 @@
+/*
+* Copyright 2018-2019 Redis Labs Ltd. and Contributors
+*
+* This file is available under the Redis Labs Source Available License Agreement
+*/
+
+#include "reduce_traversal.h"
+
+#include "../../util/arr.h"
+#include "../../util/vector.h"
+#include "../ops/op_expand_into.h"
+#include "../ops/op_conditional_traverse.h"
+#include "../ops/op_cond_var_len_traverse.h"
+
+static bool _entity_resolved(OpBase *root, const char *entity) {
+    int count = (root->modifies) ? Vector_Size(root->modifies) : 0;
+
+    for(int i = 0; i < count; i++) {
+        char *alias;
+        Vector_Get(root->modifies, i, &alias);
+        if(strcmp(entity, alias) == 0) return true;
+    }
+
+    for(int i = 0; i < root->childCount; i++) {
+        OpBase *child = root->children[i];
+        if(_entity_resolved(child, entity)) return true;
+    }
+
+    return false;
+}
+
+void _removeRedundentTraversal(ExecutionPlan *plan, CondTraverse *traverse) {
```

Comment:
Typo - should be `_removeRedundantTraversal`

Reply:
👍